src/App.tsx

```diff
@@ -1,19 +1,147 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+
+import { createTodo, deleteTodo, getTodos } from './api/todos';
+
 import { AuthContext } from './components/Auth/AuthContext';
+import { ErrorNotifications } from './components/ErrorNotifications';
+import { Filters } from './components/Filters';
+import { NewTodo } from './components/NewTodo/NewTodo';
+import { TodoList } from './components/TodoList';
+
+import { ErrorType } from './types/ErrorType';
+import { FilterType } from './types/FilterType';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>(todos);
+  const [filterBy, setFilterBy] = useState<FilterType>(FilterType.ALL);
+  const [error, setError] = useState<ErrorType>(ErrorType.NONE);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [isAdding, setIsAdding] = useState(false);
+  const [deletedTodoIds, setDeletedTodoIds] = useState<number[]>([]);
+
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const loadTodos = useCallback(async () => {
+    if (user) {
+      try {
+        const todosFromServer = await getTodos(user.id);
+
+        setTodos(todosFromServer);
+      } catch {
+        setError(ErrorType.API);
+      }
+    }
+  }, []);
+
+  const handleAddNewTodo = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (user) {
+      setIsAdding(true);
+
+      try {
+        if (!todoTitle.trim().length) {
+          setError(ErrorType.TITLE);
+          setIsAdding(false);
+
+          return;
+        }
+
+        await createTodo(todoTitle, user.id);
+        await loadTodos();
+
+        setIsAdding(false);
+        setTodoTitle('');
+      } catch {
+        setError(ErrorType.API);
```

Comment:
API isn't the best name for an error variable, it is better to name it based on the error that occurs